python/pyarrow/gandiva.pyx

```diff
@@ -97,6 +100,21 @@ cdef class Expression(_Weakrefable):
     cdef void init(self, shared_ptr[CExpression] expression):
         self.expression = expression
 
+cdef class Configuration(_Weakrefable):
+    cdef:
+        shared_ptr[CConfiguration] config
+
+    def __init__(self):
+        raise TypeError("Do not call {}'s constructor directly, use the "
+                        "TreeExprBuilder API directly"
+                        .format(self.__class__.__name__))
+
+    @staticmethod
+    cdef create():
+        cdef Configuration self = Configuration.__new__(Configuration)
+        self.config = CConfigurationBuilder.DefaultConfiguration()
+        return self
+
```

Comment:
```suggestion
```

This class is currently unnecessary since it only wraps the default configuration. Exposing it can wait until there's a need for python to adjust a Configuration.